packages/web-app-files/src/helpers/resources.js

```diff
@@ -111,16 +112,39 @@ export function buildResource(resource) {
 }
 
 export function buildSpace(space) {
-  let spaceImageData, spaceReadmeData, spacePermissions
+  let spaceImageData, spaceReadmeData
   let disabled = false
 
+  const spaceRoles = {}
+  for (const role of SpacePeopleShareRoles.list()) {
+    spaceRoles[role.name] = []
+  }
+
   if (space.special) {
     spaceImageData = space.special.find((el) => el.specialFolder.name === 'image')
     spaceReadmeData = space.special.find((el) => el.specialFolder.name === 'readme')
   }
 
   if (space.root) {
-    spacePermissions = space.root.permissions
+    for (const permission of space.root.permissions) {
+      if (permission.roles.includes(spaceRoleViewer.name)) {
```

Comment:
Please loop over the space roles, otherwise you'll still have hardcoded references of the respective roles.